packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -797,14 +771,44 @@ const VirtualListBaseFactory = (type) => {
 			delete rest.spacing;
 			delete rest.updateStatesAndBounds;
 
+			const
+				{dataSize, direction, itemSize, overhang, rtl, spacing} = this.prevProps,
+				{props} = this,
+				hasMetricsChanged = (
+					direction !== props.direction ||
+					((itemSize instanceof Object) ? (itemSize.minWidth !== props.itemSize.minWidth || itemSize.minHeight !== props.itemSize.minHeight) : itemSize !== props.itemSize) ||
+					overhang !== props.overhang ||
+					spacing !== props.spacing
+				);
+
+			this.hasDataSizeChanged = (dataSize !== props.dataSize);
+
+			if (hasMetricsChanged || this.props.clientSize && this.primary === null && this.secondary === null) {
+				this.calculateMetrics(props);
+				this.updateStatesAndBounds(props, true);
+				this.setContainerSize();
+			} else if (this.hasDataSizeChanged) {
+				this.updateStatesAndBounds(props, true);
+				this.setContainerSize();
```

Comment:
Why do we need to reset container size when only the `dataSize` has changed?

Reply:
I just copied my comment in https://github.com/enactjs/enact/pull/2108#discussion_r257104333.
```
In addition, when metrics changes or data size changes, we need to update container size again because the visibility of the scrollbar could changes and the scroll area could changes.
```

Reply:
<On Feb 18th>
- I updated the PR.